### PR TITLE
feat: Add new exit events for token predicates

### PR DESCRIPTION
**Pull request by QEDK**
Adds new exit events for all token predicates similar to `ExitedEther` No other functional change is raised in this PR.

### CODE REVIEW

**Comment by QEDK** [comment]
> Contracts look good, lets add some test cases.

Tests are added, @imyourm8 was able to run them locally, any ideas about fixing the CI?

**Review by maxsam4 — CHANGES_REQUESTED** [review]
Why have the flat contracts changed so much? It seems like a newer version of OZ was used to generate these. We should pin the version of OZ and use the same one for this PR. Can upgrade OZ version in a new PR, if needed.

ps The core changes look good, I just don't want so many changes in flat contracts to be part of this PR because it's easy to miss a typo in the mountain of unnecessary changes.

**Comment by QEDK** [comment]
> Why have the flat contracts changed so much? It seems like a newer version of OZ was used to generate these. We should pin the version of OZ and use the same one for this PR. Can upgrade OZ version in a new PR, if needed.
> 
> ps The core changes look good, I just don't want so many changes in flat contracts to be part of this PR because it's easy to miss a typo in the mountain of unnecessary changes.



> Why have the flat contracts changed so much? It seems like a newer version of OZ was used to generate these. We should pin the version of OZ and use the same one for this PR. Can upgrade OZ version in a new PR, if needed.
> 
> ps The core changes look good, I just don't want so many changes in flat contracts to be part of this PR because it's easy to miss a typo in the mountain of unnecessary changes.

This is fixed, but it seems that the flattener is working slightly differently as well so some newlines changes.

File: flat/BaseRootTunnel.sol [hunk 1]
```diff
@@ -1670,9 +1663,7 @@ abstract contract BaseRootTunnel is AccessControlMixin {
         stateSender.syncState(childTunnel, message);
     }
 
-    function _validateAndExtractMessage(bytes memory inputData) internal returns (bytes memory) {
-        require(inputData.length == 10, "RootTunnel: BAD_PAYLOAD");
```
Comment (maxsam4) [inline flat/BaseRootTunnel.sol:1674]: Why is this gone?

File: flat/RootTunnel.sol [hunk 2]
```diff
@@ -1670,9 +1663,7 @@ abstract contract BaseRootTunnel is AccessControlMixin {
         stateSender.syncState(childTunnel, message);
     }
 
-    function _validateAndExtractMessage(bytes memory inputData) internal returns (bytes memory) {
-        require(inputData.length == 10, "RootTunnel: BAD_PAYLOAD");
```
Comment (maxsam4) [inline flat/RootTunnel.sol:1674]: Why is this gone?

File: flat/BaseRootTunnel.sol (hunk 1 above)
Comment (QEDK) [reply flat/BaseRootTunnel.sol:1674]: This is a change rebased from `master` 88dbf0a88fd68fa11f7a3b9d36629930f6b93a05

**Review by maxsam4 — APPROVED** [review]

**Review by itzmeanjan — APPROVED** [review]
Looks fine

**Pull request merged**